### ISSUE TITLE
Add Lint/RescueWithoutErrorClass

### ISSUE THREAD

**Issue by no-reply** [samvera-labs/bixby#23]
See: https://github.com/samvera/hyrax/pull/2655#discussion_r168789021

**Comment by no-reply** [comment]
Actually, we want `Style/RescueStandardError`.

**Comment by no-reply** [comment]
I expressed some concerns about the `Lint/RescueWithoutErrorClass` rule here: https://github.com/samvera/hyrax/pull/2655#discussion_r168802131

To elaborate. I think it's a pretty common use case for some code (e.g. a Controller) to catch all errors, do some work (log, add a flash message, etc...), and/or reraise with a localized error type.

That cop has been removed, and replaced with `Style/RescueStandardError`. By default, this cop ensures that `rescue StandardError` is explicitly specified. My sense is that this is a positive development, and that we should enable the new cop.
